crates/db_views/src/comment_view.rs

```diff
@@ -202,7 +212,11 @@ fn queries<'a>() -> Queries<
       .unwrap_or(false);
     // only show removed comments to admin when viewing user profile
     if !(options.is_profile_view && is_admin) {
```

Comment:
I'm wondering if it makes sense to only show removed comments to an admin if they're looking at the profile view. I would imagine it would be useful to for an admin to see the comments in the context of the post they were removed from.

In addition, I believe this logic still hides deleted comments from non-admin moderators.

Reply:
Thats true. Problem is that its very annoying when you browse the site and see offensive, removed comments everywhere. I guess ideally removed comments would be returned by the API, but the ui would hide them by default, with a button "click to view".

The profile view doesnt belong to any community, so theres no good way to show it to mods there (unless you show it to anyone who moderates any community).

Reply:
Agree, this should be left up to the UIs, and we shouldn't be messing with comment trees.

Reply:
@dessalines Just to make sure we're on the same page, you don't want to filter deleted or removed comments at all on the backend and leave that decision of how to display them up to the UI?

Reply:
Correct. At least in the case of deleted and removed, that info is directly on the `CommentView`, so UI's can just choose how they want to render it, while still not messing with the rest of the tree.

The blocked communities / persons/ instances one tho... I'm honestly not sure. My gut reaction is that those should stay the way they are, even if it does wipe out branches of the tree ( in the case of person and instance blocks). But I'm open to either way.

Reply:
Its better if the backend handles as much as possible, so its easier to implement frontends without having to handle tons of things.

If we do return deleted/removed comments from the api, we need to go back to manually setting the content to empty string like we did in the past. Also it depends on the view, there is no reason to show deleted comments on the frontpage comments tab.

[Here](https://github.com/LemmyNet/lemmy/pull/3286) is the previous PR which changed how deleted/removed comments are handled. Please give it a look so we avoid going back and forth between the same changes.

Reply:
I could roll back to the changes I have [here](https://github.com/LemmyNet/lemmy/pull/3965/commits/f93d08a147ab22d1685679b31174b78199cff5f1). This changes the logic from what it was before so that deleted/removed comments will show up if they have children or if they belong to the community the user making the request moderates.

Reply:
Well, I do prefer that it would return something for blocked comments. So the UI can inform the user, that the reason they can't view this this comment because the creator is blocked. (And if they wish to see to comment anyway they could override it)

I do not think its possible for the server to insure that it always returns complete trees. And thus the clients will have to handle this edgecase regardless of these changes.

Changing it so that the server still fully returns deleted data (and thus keeps that data). Might have some bad implications, I doubt admins would still want to store and return illegal content. Think of CSAM, but this could be anything really that is declared illegal, and which could bring them into legal trouble. Or to conform to GDPR, where a user removes his comment it should be completely gone. To be in compliance with GDPR, see "The right to erasure" Articles 17 & 19 of GDPR. All though I am no legal expert.

Other cases might be, what happens to comments from deleted users. I am not sure what we currently do, but I think the current approach is to completely remove them?

Or what about the case, where the admins straight up delete entries/comments from within the DB. (This happened during the CSAM scare)

The solution, I have now is not ideal and a bit more complex. But I do think is necessary. Maybe the server can return these "fake" missing comments to offload the complexity from the clients. But I rather not if this has a performance impact.

Reply:
Purged content (as in DB deleted), shouldn't be an issue, because a purge of a comment will wipe out all its children (IE the entire tree branch).